### PR TITLE
Fix: Don't fail on domains with no records

### PR DESCRIPTION
**Pull request by mrueg**
FYI @webwurst @aellwein
This hopefully fixes the issue, I still need to test it tomorrow with an empty domain (and might change it to logLevel debug depending on how noisy it is)

Fixes: https://github.com/mrueg/external-dns-netcup-webhook/issues/5

### CODE REVIEW

File: provider/netcup.go [hunk 1]
```diff
@@ -186,7 +190,11 @@ func (p *NetcupProvider) ApplyChanges(ctx context.Context, changes *plan.Changes
 		// Gather records from API to extract the record ID which is necessary for updating/deleting the record
 		recs, err := p.session.InfoDnsRecords(zoneName)
 		if err != nil {
-			_ = level.Error(p.logger).Log("msg", "unable to get DNS records for domain", "zone", zoneName, "error", err)
+			if p.session.LastResponse.Status == string(nc.StatusError) && p.session.LastResponse.StatusCode == 5029 {
```
Comment (aellwein) [inline provider/netcup.go:193]: LGTM, i would maybe check additionally for ``if p.session.LastResponse != nil``, just not to assume that my client implementation would always be correct to set this 🤣

Comment (mrueg) [reply provider/netcup.go]: Good point :D thanks!
